ui/src/app/pages/analysis/runtime-panel.tsx

```diff
@@ -0,0 +1,207 @@
+      const isDataproc = !!runtime.dataprocConfig;
+
+      let masterMachineName;
+      let masterDiskSize;
+      if (isDataproc) {
+        masterMachineName = runtime.dataprocConfig.masterMachineType;
+        masterDiskSize = runtime.dataprocConfig.masterDiskSize;
+      } else {
+        masterMachineName = runtime.gceConfig.machineType;
+        masterDiskSize = runtime.gceConfig.bootDiskSize;
+      }
+      const machineType = machineTypes.find(({name}) => name === masterMachineName) || defaultMachineType;
+
+      return <div data-test-id='runtime-panel'>
+        <h3 style={styles.sectionHeader}>Cloud analysis environment</h3>
+        <div>
+          Your analysis environment consists of an application and compute resources.
+          Your cloud environment is unique to this workspace and not shared with other users.
+        </div>
+        {/* TODO(RW-5419): Cost estimates go here. */}
+        <div style={styles.controlSection}>
+          {/* Recommended runtime: pick from default templates or change the image. */}
+          <FlexRow style={{justifyContent: 'space-between', marginTop: '.5rem', marginBottom: '1rem'}}>
+            <h3 style={{...styles.sectionHeader, margin: '0'}}>Custom cloud environment</h3>
+            <PopupTrigger side='bottom'
+                          closeOnClick
+                          content={
+                            <React.Fragment>
+                              <MenuItem style={styles.presetMenuItem}>General purpose analysis</MenuItem>
+                              <MenuItem style={styles.presetMenuItem}>Genomics analysis</MenuItem>
+                            </React.Fragment>
+                          }>
+              <Clickable data-test-id='runtime-presets-menu'
+                         disabled={true}>
+                Recommended environments <ClrIcon shape='caret down'/>
+              </Clickable>
+            </PopupTrigger>
+          </FlexRow>
+          <h3 style={styles.sectionHeader}>Application configuration</h3>
+          {/* TODO(RW-5413): Populate the image list with server driven options. */}
+          <Dropdown style={{width: '100%'}}
+                    data-test-id='runtime-image-dropdown'
+                    disabled={true}
+                    options={[runtime.toolDockerImage]}
+                    value={runtime.toolDockerImage}/>
+          {/* Runtime customization: change detailed machine configuration options. */}
+          <h3 style={styles.sectionHeader}>Cloud compute profile</h3>
+          <FlexRow style={{justifyContent: 'space-between'}}>
+            <div>
+              <label htmlFor='runtime-cpu'
+                     style={{marginRight: '.25rem'}}>CPUs</label>
+              <Dropdown id='runtime-cpu'
+                        disabled={true}
+                        options={fp.flow(
+                          /* Show all CPU options. */
+                          fp.map('cpu'),
+                          fp.union([machineType.cpu]),
```

Comment:
What are you intending to do with this? As written this unions the current CPU selection with all machine types, which should just be... all machine types. I believe this will also result in duplicate CPUs. Is it possible you meant `fp.unionBy('cpu')`?

Reply:
Gotcha, that makes sense. Thank you!